Crust/Mapper/MappingOperator.swift

```diff
@@ -92,6 +104,31 @@ public func mapField<T: JSONable, C: MappingContext where T == T.ConversionType>
     return map.context
 }
 
+public func mapField<T: JSONable, C: MappingContext where T == T.ConversionType, T == T.ConversionType, T: AnyMappable>(inout field: T?, map:(key: JSONKeypath, context: C)) -> C {
+    
+    guard map.context.error == nil else {
+        return map.context
+    }
+    
+    switch map.context.dir {
+    case .ToJSON:
+        let json = map.context.json
+        map.context.json = mapToJson(json, fromField: field, viaKey: map.key)
+    case .FromJSON:
+        do {
+            if let baseJSON = map.context.json[map.key] {
+                try mapFromJson(baseJSON, toField: &field)
+            } else {
+                throw NSError(domain: CRMappingDomain, code: 0, userInfo: nil)
+            }
+        } catch let error as NSError {
+            map.context.error = error
+        }
+    }
+    
+    return map.context
+}
```

Comment:
Was this function actually required to fix the problem?

If `T: JSONable` and `T: AnyMappable` as defined here and the code here is exactly the same as the the case where `T: JSONable` by itself then that function should already be enough for this case.